buildtest/executors/base.py

```diff
@@ -0,0 +1,102 @@
+"""
+BuildExecutor: manager for test executors
+Copyright (C) 2020 Vanessa Sochat. 
+"""
+
+import os
+import sys
+
+from buildtest.config import config_opts
+from buildtest.log import init_logfile, init_log
+
+
+class BuildExecutor:
+    """A BuildExector is a base class some type of executor, defined under
+       the buildtest/settings/default-config.json schema. For example,
+       the types "local" and "slurm" would map to `LocalExecutor` and
+       `SlurmExecutor` here, each expecting a particular set of
+       variables under the config options. If options are required
+       and not provided, we exit on error. If they are optional and not
+       provided, we use reasonable defaults.
+    """
+
+    def __init__(self, config_opts):
+        """initiate executors, meaning that we provide the config_opts
+           that are validated, and can instantiate each executor to be available
+           
+           Parameters:
+
+           config_opts: the validated config opts provided by buildtest.
+        """
+        self.executors = {}
+
+        # Load the executors
+        for name, executor in config_opts.get("executors", {}).items():
+            if executor["type"] == "local":
+                self.executors[name] = LocalExecutor(name, executor)
+            elif executor["type"] == "slurm":
+                self.executors[name] = SlurmExecutor(name, executor)
+
+    def __str__(self):
+        return "[buildtest-executor]"
+
+    def __repr__(self):
+        return "[buildtest-executor]"
+
+    def get(self, name):
+        """Given the name of an executor return the executor for running 
+           a buildtest build.
+        """
+        return self.executors.get(name)
+
+
+class BaseExecutor:
+    """The BaseExecutor is an abstract base class for all executors
```

Comment:
Can we add public methods in docstring and end last sentence with a period

Reply:
Please add an automated check for this, so the CI tells me.

Reply:
do you have an automated check in mind, i would like to have something like this.

Reply:
The only place I've seen this run is for urlstechie, and possibly apache airflow. But I've never used this integration so I can't comment on the kind of setup, etc. required. https://app.codacy.com/gh/urlstechie/urlchecker-action/commits?bid=16994166